folium/plugins/locate_control.py

```diff
@@ -0,0 +1,61 @@
+"""Add Locate control to folium Map.
+
+Based on leaflet plugin: https://github.com/domoritz/leaflet-locatecontrol
+"""
+
+from branca.element import CssLink, Figure, JavascriptLink, MacroElement
+
+from jinja2 import Template
+
+
+class LocateControl(MacroElement):
+    """Control plugin to geolocate the user.
+
+    Parameters
+    ----------
+    options: dict, optional
+        For possible options, see https://github.com/domoritz/leaflet-locatecontrol
+
+    Examples
+    --------
+    >>> import folium
+    >>> from folium.plugins import LocateControl
+    >>> map = folium.Map()
+    # With default settings
+    >>> LocateControl().add_to(map)
+
+    # With custom options
+    >>> options = {"position": "topright",
+    ...             "strings":{
+    ...                     "title":"Show my current location"}}
+    >>> LocateControl(options=options).add_to(map)
+
+    For more info check:
+    https://github.com/domoritz/leaflet-locatecontrol
+
+    """
+
+    _template = Template(u"""
+                         {% macro script(this, kwargs) %}
+                         var {{this.get_name()}} = L.control.locate(
+                                        {{this.options|tojson}}).addTo({{this._parent.get_name()}});
```

Comment:
Since the options will expand, to make the rendered template more readible put the stuff after the options on the next line:
```
    {{this.options|tojson }}
).addTo(etc)
```